spec/support/matchers/have_editable_property.rb

```diff
@@ -1,18 +1,47 @@
 RSpec::Matchers.define :have_editable_property do |property, predicate|
   match do |model|
+    # @todo deprecate two-argument call style in favor chaining
+    predicate ||= chain_predicate
+    predicate = RDF::URI.intern(predicate)
+
     values = ['Comet in Moominland', 'Moomin Midwinter']
+    values = values.first if @single
+
+    config = model.class.properties.fetch(property.to_s) do
+      @missing_property = true
+      return false
+    end
 
     expect { model.public_send("#{property}=", values) }
-      .to change { model.public_send(property).to_a }
+      .to change { Array(model.public_send(property)) }
       .to contain_exactly(*values)
 
-    expect(model.resource.statements)
-      .to include(RDF::Statement(model.rdf_subject,
-                                 predicate,
-                                 'Comet in Moominland'),
-                  RDF::Statement(model.rdf_subject,
-                                 predicate,
-                                 'Moomin Midwinter'))
+    return true unless predicate
+
+    @actual_predicate = config.predicate
+    return false unless @actual_predicate == predicate
+
+    Array(values).each do |value|
+      expect(model.resource.statements)
+        .to include(RDF::Statement(model.rdf_subject, predicate, value))
+    end
+
     true
   end
+
+  chain :with_predicate, :chain_predicate
+
+  chain :as_single_valued do
+    @single = true
+  end
+
+  failure_message do |model|
+    msg = "expected #{model.class} to have an editable property `#{property}`"
+    msg += " with predicate #{predicate.to_base}." if predicate
+
+    return msg + "\n\tNo property for `#{property}` was found." if @missing_property
+
+    msg += "\n\tThe configured predicate was #{@actual_predicate.to_base}" unless @actual_predicate == predicate
+    msg
```

Comment:
In the base case error message you get:
```
expected ETD to have an editable property department
```
Based on that error message alone, I'm not sure it's clear whether the error was that the property was missing or that it wasn't editable.  From reading the matcher code, I can tell it's that latter, but would it be worth stating this explicitly in the error text:
```
expected ETD to have an editable property department
   department exists but was not editable.
```

Reply:
I think the error messages here could absolutely be improved. I stopped my extensions where I did in this case mainly because the case where the property is registered but the setters don't exist would (barring some horrific metaprogramming) be considered a bug in `ActiveTriples`. My thinking was that this much would serve well enough for this project.

My hope is to move these matchers out to some shared library (`hyrax`?) for reuse, and that they will continue to improve.

Do you think it's worth revisiting this immediately?